### DISCUSSION TITLE
Basic and extended settings

### DISCUSSION THREAD

**Discussion by SF4User** [devemux86/cruiser#344] [Q&A] [answered]
Ich liebe und hasse diese App!
Auf der einen Seite kann man sich einen schönen Desktop einrichten, auf der anderen Seite gibt es eine riesige Müllhalde an Optionen, Einstellungen, 1000 Offline Karten, 1000 Online Karten, 1000 Routendienste u.s.w.
Man wird verrückt dabei. Man hat eine schöne Einstellung  . . . dann ändert man irgendwo eine Kleinigkeit, den Render Mode ?? Und plötzlich sind die Tankstellen,  die man vorher mühsam eingestellt hatte, weg.
Gestern: Tour von A nach B, ein Shaping Point. Vorhaben erledigt, nun wieder zurück. Unbekannte Gegend. 
Also: Route umdrehen . . . ??????
Ich habe heute, nachträglich Stunden diesen verdammten Bottom gesucht: Route umdrehen. Das ist selbst im Calimoto eine one Click Feature . . .
Hier sucht man sich deppert !!!
Bitte, Devemux . . . bring bitte endlich Ordnung und Struktur in diese völlig überfrachtete App.
Für den normalen Anwender ist die App für die Tonne, eine Müllsammlung an Möglichkeiten, aber keine Motorrad App, wo man unterwegs nicht stundenlang da rumstehen kann und möchte.
Bitte aufräumen und Struktur !!!
<img src="https://github.com/devemux86/cruiser/assets/132825580/e31bfa8b-6f1d-4161-b12c-354ffe7e6546" width="50%"/>
Wer bitte, benötigt diese ganzen Routendienste?? Das ist doch alles nur verwirrend !!
Und das ist nur ein (!) Beispiel . . .

**Comment by WolfgangZ42** [comment]
> Also: Route umdrehen . . . ??????

Gar nicht schwer - wenn man's weiß: (und Du solltest es wissen - siehe [https://github.com/devemux86/cruiser/discussions/304](https://github.com/devemux86/cruiser/discussions/304))

<img width="100" src="https://github.com/devemux86/cruiser/assets/121806238/ceb2b280-e4a7-48e5-a7b9-14adfbebc5da"/>  
<img width="100" src="https://github.com/devemux86/cruiser/assets/121806238/74407537-86e0-48e3-b0c5-46dabbfcc845"/>  
<img width="100" src="https://github.com/devemux86/cruiser/assets/121806238/5d6f1264-e2c4-412a-9816-410486caa02d"/>  
<img width="100" src="https://github.com/devemux86/cruiser/assets/121806238/89bf0c35-f91c-473b-8c92-a5598f47db6c"/>

Drei Clicks ... 

> Wer bitte, benötigt diese ganzen Routendienste?? 

Es sind sogar noch mehr:

<img width="300" src="https://github.com/devemux86/cruiser/assets/121806238/4a81f2b1-f423-4dad-af48-1b0342ecae21"/>

Ich benutze ausschließlich **BRouter (offline)** und ignoriere alle anderen - kein Problem. Vergleiche es mit den annähernd 200 Fernsehsendern, die Dir bei modernen Bildschirmen angeboten werden, davon interessieren mich vielleicht 2 Dutzend - den Rest vergesse ich schlichtweg.

**Reply by SF4User** [reply]
Jaaahh ! Ich hab's vergessen ! 
Wieso aber ned systematisch ?
Burger . . . Routen . . . Und dort den Bottom: Route umdrehen. Das wäre doch nur logisch.
<img src="https://github.com/devemux86/cruiser/assets/132825580/9f4ca12a-3802-44f5-86bd-b8fd6cc6369d" width="50%"/>
Diese versteckten Funktionen sind komplett destruktiv . . . Unterwegs,  wenn man auch noch Stress hat, unbrauchbar !

**Comment by DolomitesTom** [comment]
@SF4User 
Beschäftige dich entweder mit der App und lies die Dokumenation oder benutze weiter Kurviger 3, aber verunglimpfe nicht die Cruiser App, die mit Abstand beste App, die es gibt für Motoradfahrer. 
Kurviger 3 kann verglichen mit Cruiser fast nichts, nur die rudimentäre Navigation, ist Lichtjahre hinten dran mit aus meiner Sicht essentiellen Funktionen, die nie kommen, weil der Wille fehlt. 

Devemux ist offen für Vorschläge und Verbesserungen, die Sinn machen. Einen extra Button für Route umdrehen im Route Menü? Klar, kein schlechter Vorschlag. Nur hast du oben gegen die angeblich überfrachtete App gewettert.

Was du suchst, ist ganz einfach Menü --> Karte --> Kartenebenen und dann kann man noch die Größe der Symbole und den Zoomfaktor einstellen, bei dem diese sichtbar sein sollen. Solche high sophisticated functions wirst du bei K3 immer vergebens suchen. 
Außerdem funktioniert ALLES OFFLINE, auch die Routenberechnung! Das ist ein nicht zu unterschätzendes Feature im Alpenländle oder auf vielen Regionen in Deutschland ohne Netzabdeckung.

Route umdrehen ist jetzt nichts, das man während des Fahrens regelmäßig benötigen würde, normal nur bei der Planung und da sind 3 "Klicks" absolut kommod. Die Funktion ist erst vor kurzem in das Untermenü gewandert, vorher war es eigener Menüpunkt mit 2" Klicks". Könnte man drüber abstimmen, ob das besser wäre, ich find's ok, so wie es ist.

Wie gesagt, gib nicht der App die Schuld, sondern nur dir selbst, nämlich deiner mangelnden Flexibilität, dich richtig mit der App zu beschäftigen.

Wenn du ne einfache App willst, die aus meiner Sicht und aktuell so gut wie nichts außer rudimentäres Navigieren kann, benutze weiter K3, wie schon gesagt, aber versuche nicht, Cruiser zu den anderen 0815 Apps werden zu lassen. Von Cruiser ist  jeder andere Nutzer in der Community meiner Einschätzung nach absolut begeistert. Die Weiterentwicklungen gegenüber K2 sind gigantisch und es sind noch viele heiß ersehnte Funktionen in der Development Pipeline.

**Reply by SF4User** [reply]
@DolomitesTom
Für jemanden, der sich offensichtlich 48 Stunden/täglich mit der App beschäftigt und sich hier im Forum ständig wichtig macht, interessiert mich deine Belehrung garnicht!

**Reply by devemux86** [reply]
Please everyone, let's keep the discussion about Cruiser.
Remember this is a community we are building together. :slightly_smiling_face: 

Bitte alle, lasst uns die Diskussion über Cruiser weiterführen.
Denken Sie daran, dass wir diese Gemeinschaft gemeinsam aufbauen. :slightly_smiling_face:

**Reply by DolomitesTom** [reply]
@SF4User 
Jeder ist frei darin, sich zu interessieren oder auch nicht. Wenn man sich so prollmäßig über ne App auslässt, mit der man sich offensichtlich noch nicht in ausreichendem Maß beschäftigt hat, braucht sich über Kritik an den eigenen Ausdrucksformen jedenfalls nicht zu wundern. Es gibt immer wieder Leute, die gern austeilen und großzügig, mit Ausrufezeichen hantierend, Forderungen rausposaunen anstatt sachliche Vorschläge zu machen, aber selbst keinerlei Kritik ertragen können. Wundert mich nicht, ist halt so. Wie man in den Wald rein ruft, so schallt's wieder heraus. Ist auch oft so 😂.

**Comment by devemux86** [comment]
@SF4User thanks for the suggestion!
Cruiser can be used with the default settings. Users can explore the functions as they wish.
Some users might feel confused by the many features, so we can definitely improve the user experience.
I am always open to ideas for better organisation, improvements and of course new features! :slightly_smiling_face: 

@SF4User danke für die Anregung!
Cruiser kann mit den Standardeinstellungen verwendet werden. Die Benutzer können die Funktionen nach Belieben erkunden.
Einige Benutzer könnten sich durch die vielen Funktionen verwirrt fühlen, daher können wir die Benutzererfahrung definitiv verbessern.
Ich bin immer offen für Ideen zur besseren Organisation, Verbesserungen und natürlich für neue Funktionen! :slightly_smiling_face:

**Reply by Ciutnik** [reply]
I once suggested dividing the set of settings into "basic" and "advanced", as we see in other applications (https://github.com/devemux86/cruiser/discussions/169#discussioncomment-5927829). Straight out of the box we would have a set of "basic" options and everyone could decide whether they want and need the "advanced" options. Maybe this is a good idea for the future? Especially since the application is still developing, there will probably be more settings.

Perhaps the most difficult thing in implementing this would be determining which option is still "basic" and which is already "advanced" 🙂

**Reply by devemux86** [reply]
For the app settings it seems easier. Changing the app interface is more challenging.

Also as you mentioned, more difficult would be to separate the settings to levels (2, 3),
as users have usually different requirements and expectations.

**Reply by Cer-o** [reply]
Apoyo esa opción. Soy un entusiasta usuario de Cruiser pero el uso que hago de la aplicación es muy básico. Como no me gustaría que se complicara en exceso, creo que la propuesta de @Ciutnik puede ser muy positiva.

**Reply by Ciutnik** [reply]
> Changing the app interface is more challenging

Indeed, you are right. I guess I can withdraw my proposal. After all, we already have the "Advanced..." submenu in many settings places. Everyone CAN, but doesn't have to, look there. From the user's point of view, it makes no difference whether to have two sets "Basic" and "Advanced" or only have the "Advanced..." submenu. Especially since this does not apply to everyday activities with the application, but to settings that are used very rarely, and in many cases only once. 🙂

**Reply by devemux86** [reply]
@Ciutnik

I meant that the app interface (i.e. panels) and the app settings are different things.
Changing how the app (e.g. navigation) works is more complicated.

But we can introduce user levels in app settings, e.g. normal and advanced or even 3 levels.
This should not be difficult. We just have to "hide" some settings, so users are not confused.

**Reply by Ciutnik** [reply]
OK I see. Of course, there is no point in separating the user interface into different levels of complexity. Now, it's certainly perfect for everyone – this positively distinguishes Cruiser from other navigation systems and, among other things, this is its strength! ❤️

As I wrote, I think it's roughly the same thing whether to have two sets of settings or the „Advanced...” submenu. It's just a matter of getting used to it. But 2 levels of advancement are probably enough; 3 levels may be too much to easily distinguish what is hidden where.

**Comment by 0709wiwiwi** [comment]
Simpler by adding extra complexity? 

As the original profiles offered are ideally suitable for starting up in simplicity.
Once familiar with the app, one can start tuning and build an own best profile. 
Once one masters that correctly he is sure no longer a novice Cruiser user. 
KISS

**Comment by airald** [comment]
I like the high grade of customisation, and I think is one of Cruiser goals. It alows users to see what they want. Most of its features are programmed by a user comments or suggestions. Some users will use some options and others use another different options. This is how Cruiser can serve a wide range of users.

I know that Cruiser is not easy, but "devices" or "things" with complex functions can not be. Recently, @devemux86 put effort to place the different options in its "correct" place. Sometimes doing for itself, and other times asking community to know where should be.

Profiles has been a very good tool to use Cruiser in different situations. And recently this profiles can be exported and imported easily. It allows to share it.

People who does not know how to do something, just ask, and there are many people here to help. @SF4User please ask your doubts here, sometimes a thing that is difficult for one person is something trivial to others. Even we can share a profile. All questions can help yourself and also other users that would search for his "problems" here.

Best regards.

**Comment by devemux86** [comment]
Cruiser has several settings that can really be considered advanced.
But also many other settings that users can use the defaults (e.g. all "Appearance" settings).

So ideas for possible structures of settings:
- **2 levels**: normal, advanced
  - **Normal**: hide at least the "Advanced" settings
    - "Appearance" settings: also hide to not show too many settings?
  - **Advanced**: show all settings

- **3 levels**: normal, advanced, pro
  - **Normal**: hide "Appearance" + "Advanced" settings
  - **Advanced**: hide only "Advanced" settings
  - **Pro**: show all settings

**Reply by kds112** [reply]
I would prefer not more than two levels.... And in the basic level only really basic settings... Everything else should be visible when you select the advanced level...

**Reply by umpfmpf** [reply]
I think 2 levels are enough. 
While driving, it is very useful to see only the most necessary settings. So it remains clear. 
And if I want to set something more that does not normally occur during the journey, I can switch to the extended mode.

**Reply by devemux86** [reply]
We could also show only the options of the selected router (e.g. BRouter) and hide the options of inactive routers.
This would simplify the number of settings displayed to the user.

**Reply by umpfmpf** [reply]
For me: I love to have all the useful settings, but if there are some useless ones: away with it.

**Reply by airald** [reply]
Me too, and it's very easy to access desired function using the search.

**Reply by SF4User** [reply]
@devemux86
Mein Problem sind nicht die vielen Möglichkeiten der Einstellungen . . . das ist ja gerade das Feature von Cruiser.
Gerade das liebe ich ja an Cruiser - deshalb habe ich ja K3 in die Tonne gelegt . . . manchmal sucht man, aber wenn man sein Setup hat, dann paßt es!
Mein Problem sind die vielen unterschiedlichen Karten und Routenoptionen . . . GraphHopper, BRouter, MapQuest, OpenRouteService, OSRM, Valhalla u.s.w. Wieso nicht eine (!) vernünftige Karte und eine vernünftige Routenoption, die offline und online funktioniert?
Kartenoptionen: GraphHopper, Karten, POI, BRouter - dann muß man die Regionen laden - ganz grob, und dann gehts mit den Routenoptionen weiter . . .  Welcher Kartendownload nun für welche Routenoptionen? . . . da bist du schnell bei 1000 Möglichkeiten.
1x eine andere Karte geladen und eine Routenoption mal ausgewählt . . . und plözlich ist alles anders.
Dann kann man sich mühsam zurückarbeiten zum alten Setup.
Ich beschäftige mich ja nicht 48 Stunden täglich mit einer Navi App . . . jedenfalls nicht, wenn mein Setup mal konfiguriert ist.
Dann nutze ich das Tool . . . das wäre das Ziel.

**Reply by ramondevic** [reply]
For me too, 2 options would be enough: basic and advanced.
With the basic version it would be necessary to propose a minimal and functional configuration but I don't know how it should be, maybe just to go from A to B and little else, avoiding the complication of downloading points and maps, I really don't know what I should have: (
With the advance, it would obviously be necessary to be able to stir everything, as it is now.

(translated from Catalan)

**Reply by ramondevic** [reply]
@SF4User 

No entenc el teu problema :(
Doncs si has trobat una configuració que ja et funciona bé fes una exportació dels perfils i llestos.
Si remenes per tenir millors rendiments i visualitzacions sempre podràs tornar a tal com ho tenies abans.

I don't understand your problem :(
Well, if you've found a configuration that already works well for you, export the profiles and you're done.
If you shuffle for better performance and views you can always go back to how you were before.

**Reply by devemux86** [reply]
BRouter is an external app. It requires the extra step for its installation.
c:geo includes BRouter in c:geo app, but this requires much more work.

cc @afischerdev

---

Regarding Cruiser internal functions like the map / poi downloads, we could certainly improve the user experience.
Like OsmAnd which suggests and downloads the specific map when the user zooms in on an area.

**Reply by 0709wiwiwi** [reply]
> Like OsmAnd which suggests and downloads the specific map when the user zooms in on an area. 

And : https://github.com/devemux86/cruiser/discussions/101

**Comment by DolomitesTom** [comment]
Either 2 or 3 levels are fine.

**Comment by devemux86** [comment]
And I suppose this option is better to be somewhere inside (image 1) and not constantly in the foreground (image 2)?

<img src="https://github.com/devemux86/cruiser/assets/3484020/af1296ac-22dc-471f-bca0-87bf37c6add4" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/3b55197a-ed5e-4279-bb6a-c8c77032962d" width="50%"/>

**Reply by kds112** [reply]
Image 1 is better. You can directly switch on the advanced settings if needed. And you can decide for each menu. Question: Will the selection be stored in each profile?

**Reply by devemux86** [reply]
> Will the selection be stored in each profile?

Do we need different selection per profile?
(can be either way)

**Reply by DolomitesTom** [reply]
Normally not

**Comment by devemux86** [comment]
Some ideas about what settings could be hidden when "Extended settings" is off.

- **Location**: Appearance + Advanced
- **Map**: Appearance + Advanced
- **Search**: Appearance + Advanced
- **Routing**: Speeds + Appearance
  - **BRouter**: Other + Advanced
- **Navigation**: Recalculate route + Auto zoom + Advanced
- **Voice guidance**: Distances + Advanced
- **GPS recording**: Appearance
- **Application**: Appearance + Advanced
  - **Display**: Screen margin

We can also rearrange some options to a different group so they can be visible or not.
Reminder: next Cruiser will only show the options of the selected router (e.g. BRouter).

**Reply by DolomitesTom** [reply]
Let's see how it looks

**Reply by devemux86** [reply]
<img src="https://github.com/devemux86/cruiser/assets/3484020/21f2cd00-a144-44bf-ae39-a6c09de46439" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/ceed80a9-d0af-4a69-8af2-5d172799e89c" width="33%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/36b33315-fbc5-4020-9645-ecac46a4083e" width="33%"/>

**Reply by devemux86** [reply]
<img src="https://github.com/devemux86/cruiser/assets/3484020/256f8618-f5cb-49fc-a39c-35aaf2af3fc1" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/c8e24642-e4d0-4dff-81d2-7cee9ae98dc9" width="50%"/>

**Reply by DolomitesTom** [reply]
And when the setting is switched to on 1x, all menus will appear in each category and will stay like this even if the app is closed? If so, fine.

**Reply by devemux86** [reply]
> when the setting is switched to on 1x, all menus will appear in each category and will stay like this even if the app is closed?

Correctly.

**Comment by devemux86** [comment]
On the occasion of the basic / extended settings, we could organize better some settings.
Like display panels, they can be separate for navigation and those that work everywhere.

<img src="https://github.com/devemux86/cruiser/assets/3484020/eaaf900b-44e2-4cdc-8209-fdddce104ccc" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/d5308717-c6cc-486b-9b19-f87d1d487aad" width="50%"/>

---

<img src="https://github.com/devemux86/cruiser/assets/3484020/4f236a4b-4f2e-4414-888e-763181ed85e7" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/44ba9f05-7170-4def-9b03-23154d23f588" width="50%"/>

**Reply by kds112** [reply]
Great! And changing the order...? 🤩

**Reply by devemux86** [reply]
It is still in my list. :slightly_smiling_face:

**Comment by devemux86** [comment, marked as answer]
Option for basic or extended settings is available in [Cruiser 3.0.34](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
